modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -770,16 +795,28 @@ private CompletableFuture<Object> processMultiEntryAction(ReadWriteMultiRowRepli
                         if (lockedRow != null) {
                             result.add(row);
                         } else {
-                            if (rowsToInsert.values().stream().noneMatch(row0 -> row0.keySlice().equals(row.keySlice()))) {
-                                rowsToInsert.put(new RowId(partId), row);
+                            ByteBuffer keyToCheck = row.keySlice();
+                            if (!uniqueKeys.contains(keyToCheck)) {
```

Comment:
Is this a fix for something? I see that the code is changed

Reply:
The stream-based code performed uniqueness check. I decided to replace it with HashSet-based logic to avoid overcomplicated logic inside stream's lambda.

Reply:
Ok, I think we can simplify it even further with writing `if (uniqueKeys.add(keyToCheck)) {`